docs/utils/data.rst

```diff
@@ -0,0 +1,220 @@
+   Downloading ftp://ssd.jpl.nasa.gov/pub/eph/planets/bsp/de405.bsp from https://data.nanograv.org/static/data/ephem/de405.bsp
+   |========================================|  65M/ 65M (100.00%)        19s
+
+.. _Astropy data server: http://www.astropy.org/astropy-data/
+
+Cache Management
+================
+
+Because the cache is persistent, it is possible for it to become
+inconveniently large, or become filled with no-longer-relevant data. While
+it is simply a directory on disk, each file is supposed to represent
+the contents of a URL, and many URLs do not make acceptable on-disk
+filenames. There is reason to worry that multiple astropy processes accessing
+the cache simultaneously might lead to cache corruption. The cache is
+therefore protected by a lock and indexed by a persistent dictionary
+mapping URLs to hashes of the file contents, while the file contents are
+stored in files named by their hashes. So access to the cache is easier
+with a few helpers provided by `~astropy.utils.data`.
+
+If your cache starts behaving oddly you can use
+`~astropy.utils.data.check_download_cache` to examine your cache contents and
+raise an exception if it finds any anomalies.  If a single file is undesired or
+damaged, it can be removed by calling
+`~astropy.utils.data.clear_download_cache` with an argument that is the URL it
+was obtained from, the filename of the downloaded file, or the hash of its
+contents. Should the cache ever become badly corrupted,
+`~astropy.utils.data.clear_download_cache` with no arguments will simply delete
+the whole directory, freeing the space and removing any inconsistent data. Of
+course, if you remove data using either of these tools, any processes currently
+using that data may be disrupted (or, under Windows, deleting the cache may not
+be possible until those processes terminate). So use
+`~astropy.utils.data.clear_download_cache` with care.
+
+To check the total space occupied by the cache, use
+`~astropy.utils.data.cache_total_size`. The contents of the cache can be
+listed with `~astropy.utils.data.get_cached_urls`, and the presence of a
+particular URL in the cache can be tested with
+`~astropy.utils.data.is_url_in_cache`. More general manipulations can be
+carried out using `~astropy.utils.data.cache_contents`, which returns a
+dict mapping URLs to on-disk filenames of their contents.
+
+If you want to transfer the cache to another computer, or preserve its contents
+for later use, you can use the functions `~astropy.utils.data.export_download_cache` to
+produce a zipfile listing some or all of the cache contents, and
+`~astropy.utils.data.import_download_cache` to load the astropy cache from such a
+zipfile.
+
+Using Astropy With Limited or No Internet Access
+================================================
+
+You might want to use astropy on a telescope control machine behind a strict
+firewall. Or you might be running contionuous integration on your Astropy
```

Comment:
```suggestion
firewall. Or you might be running contionuous integration (CI) on your Astropy
```

p.s. I think people dabbling in cloud computing also has a similar problem. For example, it might be cheaper to grab stuff off AWS S3 bucket within AWS EC2 instance than to download from the Internet. Not saying you have to mention that use case; I am just thinking aloud.